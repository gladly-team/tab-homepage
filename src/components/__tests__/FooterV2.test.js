/* eslint-env jest */

import React from 'react'
import { shallow, mount } from 'enzyme'
import toJson from 'enzyme-to-json'
import GoogleChrome from 'mdi-material-ui/GoogleChrome'
import MicrosoftEdge from 'mdi-material-ui/MicrosoftEdge'
import AppleSafari from 'mdi-material-ui/AppleSafari'

const mockProps = { onBeforeInstall: () => {} }
describe('FooterV2', () => {
  it('renders without error', () => {
    const FooterV2 = require('../FooterV2').default
    shallow(<FooterV2 {...mockProps} />)
  })

  it('matches expected snapshot', function () {
    const FooterV2 = require('../FooterV2').default
    const wrapper = shallow(<FooterV2 {...mockProps} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('clicking on the chrome icon sets local storage info', () => {
    const FooterV2 = require('../FooterV2').default
    const onBeforeInstallSpy = jest.fn()
    const wrapper = mount(<FooterV2 onBeforeInstall={onBeforeInstallSpy} />)
    expect(onBeforeInstallSpy).not.toHaveBeenCalled()
    wrapper.find(GoogleChrome).simulate('click')
    wrapper.update()
    expect(onBeforeInstallSpy).toHaveBeenCalled()
  })

  it('clicking on the edge icon sets local storage info', () => {
    const FooterV2 = require('../FooterV2').default
    const onBeforeInstallSpy = jest.fn()
    const wrapper = mount(<FooterV2 onBeforeInstall={onBeforeInstallSpy} />)
    expect(onBeforeInstallSpy).not.toHaveBeenCalled()
    wrapper.find(MicrosoftEdge).simulate('click')
    wrapper.update()
    expect(onBeforeInstallSpy).toHaveBeenCalled()
  })

  it('clicking on the Safari icon sets local storage info', () => {
    const FooterV2 = require('../FooterV2').default
    const onBeforeInstallSpy = jest.fn()
    const wrapper = mount(<FooterV2 onBeforeInstall={onBeforeInstallSpy} />)
    expect(onBeforeInstallSpy).not.toHaveBeenCalled()
    wrapper.find(AppleSafari).simulate('click')
    wrapper.update()
    expect(onBeforeInstallSpy).toHaveBeenCalled()
  })
})
