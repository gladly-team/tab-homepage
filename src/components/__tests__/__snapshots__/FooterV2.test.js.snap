// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FooterV2 matches expected snapshot 1`] = `
<div
  className="makeStyles-backgroundContainer-2"
>
  <div
    className="makeStyles-contentRow-3"
  >
    <div
      className="makeStyles-columnOne-4"
    >
      <Link
        hoverStyle={Object {}}
        style={Object {}}
        to="/"
      >
        <img
          src="test-file-stub"
          style={
            Object {
              "height": 43,
            }
          }
        />
      </Link>
      <WithStyles(ForwardRef(Typography))
        color="textSecondary"
        variant="subtitle2"
      >
        Social
      </WithStyles(ForwardRef(Typography))>
      <div
        className="makeStyles-iconRow-7"
      >
        <Link
          hoverStyle={Object {}}
          style={Object {}}
          to="https://www.facebook.com/TabForACause"
        >
          <WithStyles(ForwardRef(IconButton))
            classes={
              Object {
                "root": "makeStyles-iconButton-9",
              }
            }
            edge="start"
          >
            <Memo(ForwardRef(FacebookIcon))
              classes={
                Object {
                  "root": "makeStyles-iconStyles-8",
                }
              }
            />
          </WithStyles(ForwardRef(IconButton))>
        </Link>
        <Link
          hoverStyle={Object {}}
          style={Object {}}
          to="https://www.instagram.com/tabforacause/"
        >
          <WithStyles(ForwardRef(IconButton))
            classes={
              Object {
                "root": "makeStyles-iconButton-9",
              }
            }
          >
            <Memo(ForwardRef(InstagramIcon))
              classes={
                Object {
                  "root": "makeStyles-iconStyles-8",
                }
              }
            />
          </WithStyles(ForwardRef(IconButton))>
        </Link>
        <Link
          hoverStyle={Object {}}
          style={Object {}}
          to="https://twitter.com/TabForACause"
        >
          <WithStyles(ForwardRef(IconButton))
            classes={
              Object {
                "root": "makeStyles-iconButton-9",
              }
            }
          >
            <Memo(ForwardRef(TwitterIcon))
              classes={
                Object {
                  "root": "makeStyles-iconStyles-8",
                }
              }
            />
          </WithStyles(ForwardRef(IconButton))>
        </Link>
        <Link
          hoverStyle={Object {}}
          style={Object {}}
          to="https://www.tiktok.com/@tabforacause.official?"
        >
          <WithStyles(ForwardRef(IconButton))
            classes={
              Object {
                "root": "makeStyles-iconButton-9",
              }
            }
          >
            <WithStyles(ForwardRef(SvgIcon))
              style={
                Object {
                  "fill": "#fff",
                  "height": "38px",
                  "marginTop": "-3px",
                  "width": "38px",
                }
              }
              viewBox="0 0 50 50"
            >
              <path
                d="M41,4H9C6.243,4,4,6.243,4,9v32c0,2.757,2.243,5,5,5h32c2.757,0,5-2.243,5-5V9C46,6.243,43.757,4,41,4z M37.006,22.323 c-0.227,0.021-0.457,0.035-0.69,0.035c-2.623,0-4.928-1.349-6.269-3.388c0,5.349,0,11.435,0,11.537c0,4.709-3.818,8.527-8.527,8.527 s-8.527-3.818-8.527-8.527s3.818-8.527,8.527-8.527c0.178,0,0.352,0.016,0.527,0.027v4.202c-0.175-0.021-0.347-0.053-0.527-0.053 c-2.404,0-4.352,1.948-4.352,4.352s1.948,4.352,4.352,4.352s4.527-1.894,4.527-4.298c0-0.095,0.042-19.594,0.042-19.594h4.016 c0.378,3.591,3.277,6.425,6.901,6.685V22.323z"
              />
            </WithStyles(ForwardRef(SvgIcon))>
          </WithStyles(ForwardRef(IconButton))>
        </Link>
      </div>
      <WithStyles(ForwardRef(Typography))
        color="textSecondary"
        variant="subtitle2"
      >
        Browser Compatibility
      </WithStyles(ForwardRef(Typography))>
      <div
        className="makeStyles-iconRow-7"
      >
        <WithStyles(ForwardRef(IconButton))
          classes={
            Object {
              "root": "makeStyles-iconButton-9",
            }
          }
          edge="start"
          onClick={[Function]}
        >
          <Memo(ForwardRef)
            className="makeStyles-iconStyles-8"
          />
        </WithStyles(ForwardRef(IconButton))>
        <WithStyles(ForwardRef(IconButton))
          classes={
            Object {
              "root": "makeStyles-iconButton-9",
            }
          }
          onClick={[Function]}
        >
          <Memo(ForwardRef)
            className="makeStyles-iconStyles-8"
          />
        </WithStyles(ForwardRef(IconButton))>
        <WithStyles(ForwardRef(IconButton))
          classes={
            Object {
              "root": "makeStyles-iconButton-9",
            }
          }
          onClick={[Function]}
        >
          <Memo(ForwardRef)
            className="makeStyles-iconStyles-8"
          />
        </WithStyles(ForwardRef(IconButton))>
      </div>
    </div>
    <div
      className="makeStyles-columnTwo-5"
    >
      <div
        className="makeStyles-subColumn-6"
      >
        <Link
          hoverStyle={Object {}}
          style={Object {}}
          to="https://gladly.zendesk.com/hc/en-us/categories/201939608-Tab-for-a-Cause"
        >
          <WithStyles(ForwardRef(Typography))
            color="textSecondary"
            gutterBottom={true}
            variant="subtitle2"
          >
            Help Center
          </WithStyles(ForwardRef(Typography))>
        </Link>
        <Link
          hoverStyle={Object {}}
          style={Object {}}
          to="/financials/"
        >
          <WithStyles(ForwardRef(Typography))
            color="textSecondary"
            gutterBottom={true}
            variant="subtitle2"
          >
            Financials
          </WithStyles(ForwardRef(Typography))>
        </Link>
        <Link
          hoverStyle={Object {}}
          style={Object {}}
          to="/privacy/"
        >
          <WithStyles(ForwardRef(Typography))
            color="textSecondary"
            gutterBottom={true}
            variant="subtitle2"
          >
            Privacy
          </WithStyles(ForwardRef(Typography))>
        </Link>
        <Link
          hoverStyle={Object {}}
          style={Object {}}
          to="/terms/"
        >
          <WithStyles(ForwardRef(Typography))
            color="textSecondary"
            gutterBottom={true}
            variant="subtitle2"
          >
            Terms & Conditions
          </WithStyles(ForwardRef(Typography))>
        </Link>
        <Link
          hoverStyle={Object {}}
          style={Object {}}
          to="/adblockers/"
        >
          <WithStyles(ForwardRef(Typography))
            color="textSecondary"
            gutterBottom={true}
            variant="subtitle2"
          >
            Ad Blockers
          </WithStyles(ForwardRef(Typography))>
        </Link>
      </div>
      <div
        className="makeStyles-subColumn-6"
      >
        <Link
          hoverStyle={Object {}}
          style={Object {}}
          to="/team/"
        >
          <WithStyles(ForwardRef(Typography))
            color="textSecondary"
            gutterBottom={true}
            variant="subtitle2"
          >
            Team
          </WithStyles(ForwardRef(Typography))>
        </Link>
        <Link
          hoverStyle={Object {}}
          style={Object {}}
          to="/contact/"
        >
          <WithStyles(ForwardRef(Typography))
            color="textSecondary"
            gutterBottom={true}
            variant="subtitle2"
          >
            Contact
          </WithStyles(ForwardRef(Typography))>
        </Link>
        <Link
          hoverStyle={Object {}}
          style={Object {}}
          to="/jobs/"
        >
          <WithStyles(ForwardRef(Typography))
            color="textSecondary"
            gutterBottom={true}
            variant="subtitle2"
          >
            Jobs
          </WithStyles(ForwardRef(Typography))>
        </Link>
      </div>
    </div>
  </div>
</div>
`;
